### ISSUE TITLE
UnistylesRuntime.updateTheme doesn't trigger rerender on React Native Web

### ISSUE THREAD

**Issue by mobily** [jpudysz/react-native-unistyles#179]
### Description

Hello @jpudysz! 👋 I'm rewriting my library ([Stacks](https://github.com/grapp-dev/stacks)) to TypeScript with Unistyles as a core dependency. It works perfectly, and it's amazing! ❤️ (I wanted to add variants support to Stacks as well, but you did it much better 😅)

But straight to the point, I think I have found a bug, `UnistylesRuntime.updateTheme` doesn't trigger any rerender on RNW. I have already read Unistyles source code (not deeply, honestly) and couldn't find a reason for that bug. Take a look at the screen recordings below:

Native:

https://github.com/jpudysz/react-native-unistyles/assets/1467712/8eee97dd-ba48-40c3-af12-f87302400756

Web:

https://github.com/jpudysz/react-native-unistyles/assets/1467712/22555376-c66a-4479-bd5e-2b59d3fab2b3

### Steps to reproduce

Initialize Unistyles:

```ts
/* eslint-disable @typescript-eslint/no-empty-interface */
import { UnistylesRegistry } from 'react-native-unistyles';

const breakpoints = {
  mobile: 0,
  tablet: 768,
  desktop: 998,
} as const;

const theme = {
  stacks: {
    spacing: 4,
    debug: false,
  },
};

type Breakpoints = typeof breakpoints;

type Themes = {
  readonly light: typeof theme;
};

declare module 'react-native-unistyles' {
  export interface UnistylesBreakpoints extends Breakpoints {}
  export interface UnistylesThemes extends Themes {}
}

UnistylesRegistry.addBreakpoints(breakpoints)
  .addThemes({
    light: theme,
  })
  .addConfig({
    adaptiveThemes: true,
  });
```

Add the handler to update `stacks` properties:

```tsx
const handleDebugMode = React.useCallback(() => {
  UnistylesRuntime.updateTheme('light', theme => {
    console.log(theme);
    return {
      ...theme,
      stacks: {
        ...theme.stacks,
        debug: !theme.stacks.debug,
      },
    };
  });
}, []);

// and then
<Button title="debug" onPress={handleDebugMode} />
```

### Snack or a link to a repository (optional)

https://github.com/grapp-dev/stacks/tree/main/example

### Unistyles version

2.5.3

### React Native version

0.73.6

### Platforms

React Native Web

### Engine

Hermes

### Architecture

Fabric (new)

**Comment by jpudysz** [comment]
Thanks for the report, I will check it out!

**Comment by jpudysz** [comment]
Fixed in `2.5.5`
